ebs/error_budget.py

```diff
@@ -417,6 +410,407 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
+        for key \
+             in self.exosims_pars_dict['starlightSuppressionSystems'][0] \
+                 .keys():
+            if key in dir(self):
+                setattr(self, key
+                        , self.exosims_pars_dict['starlightSuppressionSystems']
+                                                [0][key])
+
+    def initialize_walkers(self):
+        center = []
+        [center.append(np.array(val).ravel())  
+         for var_name in self.config['mcmc']['variables'] 
+         for val in self.config['mcmc']['variables'][var_name]['ini_pars']
+                               ['center']]
+        center = np.concatenate(center)
+        center = center[np.where(np.isfinite(center))]
+        spread = []
+        [spread.append(np.array(val).ravel())  
+         for var_name in self.config['mcmc']['variables'] 
+         for val in self.config['mcmc']['variables'][var_name]['ini_pars']
+                               ['spread']]
+        spread = np.concatenate(spread)
+        spread= spread[np.where(np.isfinite(spread))]
+        ndim = center.shape[0]
+        nwalkers = self.config['mcmc']['nwalkers']
+        walker_pos = center + np.random.uniform(-spread/2.0, spread/2.0
+                                                , (nwalkers, ndim))
+        return walker_pos
+
+    def update_attributes(self, values):
+        for var_name in self.config['mcmc']['variables']:
+            if var_name in dir(self):
+                template = np.array(self.config['mcmc']['variables'][var_name]
+                                               ['ini_pars']['center'])
+                indices = np.where(np.isfinite(template))
+                use_values, values = np.split(values, [indices[0].size])
+                arr = getattr(self, var_name)
+                if type(arr) == float or type(arr)==np.float64:
+                    arr = use_values[0]
+                else:
+                    arr[indices] = use_values
+                setattr(self, var_name, arr)
+                if var_name in ['contrast', 'wfe', 'wfsc_factor'
+                                , 'sensitivity']:
+                    if var_name == 'contrast':
+                        self.write_csv(var_name)
+                        self.exosims_pars_dict['starlightSuppressionSystems']\
+                                [0]['core_contrast'] = self.contrast_filename
+                    self.write_ppFact_fits()
+                    self.exosims_pars_dict['ppFact'] = self.ppFact_filename
+                if var_name == 'throughput':
+                    self.write_csv(var_name)
+                    self.exosims_pars_dict['starlightSuppressionSystems']\
+                            [0]['core_thruput'] = self.throughput_filename
+            else:
+                print(key+" not found in attributes list")
+
+    def log_prior(self, values):
+        joint_prob = 0.0
+        counter = 0
+        for i, var_name in enumerate(self.config['mcmc']['variables'].keys()):
```

Comment:
I would define a class attribute 

`self.mcmc_config = self.config['mcmc']` 

or even also 

`self.mcmc_variables = self.config['mcmc']['variables']`

to save you some space here and elsewhere further up in the code